lib/agent-store.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+const Agent = require('agentkeepalive');
+const { SOCKET_TTL } = require('./constants');
+
+class AgentStore {
+    constructor() {
+        const defaultAgentOpts = { socketActiveTTL: SOCKET_TTL };
```

Comment:
The socket ttl should be configurable

Reply:
Hmmm, I was thinking the same as well, Plus have to come up with custom config options for circuit breaker pool settings + metrics + caching of fragments.. 

I will come up with some proposal.

Reply:
Thought for a while about this and I think we can customize all of the logic in a client library like perron(or create one specific for tailor - or give PR to perron to make it work). Perron already supports circuit breaker logic for clients and we can easy add cache specific logic as well..

I will give PR to perron once i have sometime to work on some features.